package/MDAnalysis/coordinates/core.py

```diff
@@ -78,10 +78,14 @@ def reader(filename, **kwargs):
     if isinstance(filename, tuple):
         Reader = get_reader_for(filename[0],
                                 format=filename[1])
-        return Reader(filename[0], **kwargs)
+        filename = filename[0]
     else:
-        Reader = get_reader_for(filename)
+        Reader = get_reader_for(filename, format=format)
+    try:
         return Reader(filename, **kwargs)
+    except ValueError:
```

Comment:
[Below](https://github.com/MDAnalysis/mdanalysis/pull/2335/files#diff-9b11c15d9c9bc0a0353da4ca6b2d456aL597) we raise a `TypeError` instead of the `ValueError` here. One can debate if TypeError is the right choice but for a start, it should at least be consistent. So change to `TypeError` here and if you think `ValueError` is better raise an issue and we address it separately in a consistent manner.